src/main/java/com/example/taxibookingapplication/service/TaxiService.java

```diff
@@ -57,24 +59,42 @@ public void registerTaxi (Taxi taxi){
 
         for(int i=0; i< taxiList.size();i++)
         {
-            if(taxiList.get(i).getStatus().equals(Status.OCCUPIED))
+            if(taxiList.get(i).getStatus().equals(OCCUPIED))
                 occupiedList.add(taxiList.get(i));
         }
         return occupiedList;
     }
 
-    public String bookTaxi(Integer taxiId){
-       Taxi taxi =  taxiRepository.findById(taxiId).get();
+    public String bookTaxi(Integer taxiId) {
+        Taxi taxi = taxiRepository.findById(taxiId).get();
+        if (taxi.getStatus().equals(AVAILABLE)) {
 
-       taxi.setStatus(Status.OCCUPIED);
+        taxi.setStatus(OCCUPIED);
 
-       taxiRepository.save(taxi);
-
-       return taxi.getTaxiNumber();
+        taxiRepository.save(taxi);
 
+        return taxi.getTaxiNumber();
+    }
+    else
+        throw getTaxiIdNotFoundException(taxiId);
```

Comment:
Make this a more meaningful exception type. Kind of InappropriateTaxiStatusException. Irrespective of the implementation the name must be changed.

Reply:
Need to handle in controller level.